test-data/unit/check-flags.test

```diff
@@ -963,6 +963,12 @@ from missing import Unchecked
 
 t: Unchecked = 12  # E: Type of variable becomes "Any" due to an unfollowed import
 
+[case testAllowImplicitAnyVariableDefinition]
```

Comment:
I'm not fully sure how to test explicitly setting what is already the default behaviour. This test will pass whether or not `--allow-any-unimported` is set (as long as `--disallow-any-unimported` is not set).

This is a copy of the test immediately before it. The test before this sets `--disallow-any-unimported` and asserts that an error occurs, this test sets `--allow-any-unimported` and asserts no error occurs.